### DISCUSSION TITLE
Some feature requests

### DISCUSSION THREAD

**Discussion by shawmik7** [namidaco/namida#280] [Ideas]
This is a great app, you are doing a great job there! However, this app still needs to implement some more small features to make some things complete.

1. Of course there is a dedicated lyric view, but it would be great to show the currently playing line of the synced lyric on the now playing screen (you call it mini player, i guess), when the lyric view is off. See these screenshots of the "Vanilla Music Player", currently playing the line "Here comes the rain again":
![Screenshot_20240714-171250_Photos.png](https://github.com/user-attachments/assets/d9bd5b9f-38c6-44fb-b759-9f61e932f9e8)

![Screenshot_20240714-170812_Vanilla.png](https://github.com/user-attachments/assets/cc127a90-1b62-4bcc-b7dd-03b581d16b01)

2. When I open the "lyrics search view" and search for lyrics, add there an option to save that as a .lrc file in the same directory as the song itself (and of course with the same name as the song)

3. Add the ability to select which lyric I want to see when playing a song, (the embedded one or the .lrc file) in the "lyrics search view", so I can see my preferred lyrics for each song.
(Another method could be something like: pressing the lyrics button toggles between embedded lyrics, .lrc file and turn off the lyric view)

4. Add the ability to edit the .lrc file, not just the embedded one.
(You can add another field in the tag editor screen that shows the .lrc file, from where we can edit it along with the embedded one)

5. When we edit some metadata or tags and press the back button, add an auto refresh feature, so that I don't have to restart the app in order to see the updated info (This is needed if I have to edit the synced lyric for some reason, and if I have to restart the app each time to check if my timing is correct, well, that's a pain)

7. Add an option in the tag editor to fetch metadata from various sources (like spotify and somewhere else) and apply them in the song.

6. This is optional, but you could add a feature to make synced lyrics. You can check out the "Musicolet" app for this. It has this feature in the lyrics editor view.

I can think of these features as of now, since I found out about this great music player just yesterday, and instantly fell in love with it. I would fell even more if you could implement these features...

Thanks and take love 💖

**Comment by Talha1788** [comment]
Yeah, I also want the lyrics file creating, editing and syncing feature. That will be so goddamn helpful.

**Reply by muhammadrummankhan** [reply]
yeah much needed feature, there's already open source lrc editor, they can maybe add that in namida

**Reply by MSOB7YY** [reply]
@muhammadrummankhan which one? we cant just add random editor, we would need to manually design the ui and write the functions for it

**Comment by MSOB7YY** [comment]
thanks a lot for these suggestions!
1. quite nice idea but is there even any place left?
2. a similar request here #173, but its quite complex to allow that
3. that would require a separate database to save ur preference, i suggest keeping ur favourite lyrics in track's lyrics metadata, and enable "prioritize embedded lyrics"
4. lrc editor is also requested 
5. metadata is always auto refreshed after editing. if you edit lyrics tag while lyrics view is enabled, u might need to disable and enable again 
6. requested at #5
7. this would be the same as 4

thank u so much again and these ideas are definitely considered! <3
